app-router/installation.mdx

```diff
@@ -0,0 +1,168 @@
+
+```sh
+MAKESWIFT_SITE_API_KEY=paste-your-api-key-here
+```
+
+  </Step>
+  <Step title="Add Makeswift runtime and client">
+  
+Create files for the Makeswift [runtime](/reference/runtime/constructor) and [client](/reference/client/constructor).
+
+<CodeGroup>
+  <RuntimeExample />
+  <ClientExample />
+</CodeGroup>
+
+</Step>
+  <Step title="Add the Makeswift API handler">
+  
+Similar to [NextAuth.js](https://next-auth.js.org/), Makeswift uses an API handler to communicate with your Next.js app. Create the file `app/api/makeswift/[...makeswift]/route.ts`. It's important that this directory and file structure is followed precisely.
+
+<ApiHandlerExample />
+
+This API route adds support for
+[preview Mode](https://nextjs.org/docs/pages/building-your-application/configuring/preview-mode),
+[on-demand revalidation](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation),
+and other features that make Makeswift work seamlessly with your Next.js app.
+
+  </Step>
+  <Step title="Add the Next.js plugin">
+  
+Next.js plugins are configured in the project's next.config.js file by wrapping `nextConfig`. The Makeswift Next.js plugin whitelists Makeswift image domains and sets up rewrites to enable preview mode in the Makeswift builder.
+
+<NextConfigExample />
+
+  </Step>
+  <Step title="Create Makeswift provider component">
+    Create a client component for the Makeswift providers.
+
+    <ProvidersExample />
+
+  </Step>
+  <Step title="Update the root layout">
+    Wrap your app with the `MakeswiftProviders` component created in the last step and add the `DraftModeScript` to the `head` of your root layout. This script enables draft mode when inside the Makeswift builder.
+
+    <LayoutExample />
+
+  </Step>
+  <Step title="Register components with Makeswift">
+  
+Create a file for registered components called `makeswift/components.tsx`. In this example, only one component is registered. However, as you register more components, we recommend creating separate files for each component and rolling up the imports in the `makeswift/components.ts` file. Learn more about [registering components](/reference/runtime/register-component).
+
+```tsx makeswift/components.tsx
+import { ReactRuntime } from "@makeswift/runtime/react"
+import { Style } from "@makeswift/runtime/controls"
+
+function HelloWorld(props) {
+  return <p {...props}>Hello, world!</p>
+}
+
+ReactRuntime.registerComponent(HelloWorld, {
+  type: "hello-world",
+  label: "Hello, world!",
+  props: {
+    className: Style(),
+  },
+})
+```
+
+  </Step>
+  <Step title="Add a route for Makeswift pages">
+  
+Create an [optional catch-all route](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) named `[[...path]]`. You will use the catch-all route to fetch page snapshots from the `Makeswift` client and pass them to the [`Page`](/reference/components/page) component to render.
```

Comment:
> You will use the catch-all route to fetch page snapshots from the `Makeswift` client and pass them to the [`Page`](/reference/components/page) component to render. 

-> this has some jargon that I wouldn't expect all devs to know maybe something like: 

 This catch-all route will fetch page data from Makeswift and pass it to be rendered in the [`Page`](/reference/components/page) component.

Reply:
✅